diff --git a/src/core/ext/xds/xds_listener.h b/src/core/ext/xds/xds_listener.h
index 61427483f1..a3bfed0780 100644
--- a/src/core/ext/xds/xds_listener.h
+++ b/src/core/ext/xds/xds_listener.h
@@ -79,6 +79,8 @@ struct XdsListenerResource : public XdsResourceType::ResourceData {
   };
 
   struct DownstreamTlsContext {
+    DownstreamTlsContext() {}
+
     CommonTlsContext common_tls_context;
     bool require_client_certificate = false;
 
